### PR TITLE
Upgrade 13

### LINKED ISSUE #51: add #65 Upgrade to agoric-upgrade-13
working on a dapp, somebody noted auto-provisioning is now on mainnet, so it should be here now...

ref:
 - [#65 Upgrade to agoric-upgrade-13](https://ping.pub/agoric/gov/65) Voting end 2023-12-20

### PR DESCRIPTION
**Pull request by mhofman**
Closes #51 

Adds a proposal for the upcoming upgrade-13. Using a proposal number as I need this to be included in the tag generated for the PR to get https://github.com/Agoric/agoric-sdk/pull/8593 passing.

Includes a couple refactors to move common helpers into the shared library.

### CODE REVIEW

**Review by turadg — COMMENTED** [review]
> Adds a proposal for the upcoming upgrade-13.

To get a number in this repo's main, the proposal must have passed in https://bigdipper.live/agoric/proposals

I'll review again once the proposal has passed. 

> Using a proposal number as I need this to be included in the tag generated for the PR to get https://github.com/Agoric/agoric-sdk/pull/8593 passing.

I take it the production of the PR's image means you're not blocked on this being in master.

> Includes a couple refactors to move common helpers into the shared library.

Given how this build is factored now, those changes invalidate the base image. I'm reluctant to accept them because it will cause each developer's machine to have to rebuild for an hour to get them. This is part of the motivation for proposal package imports versioning in https://github.com/Agoric/agoric-3-proposals/pull/46/

**Review by turadg — APPROVED** [review]

**Comment by mhofman** [comment]
> Given how this build is factored now, those changes invalidate the base image. I'm reluctant to accept them because it will cause each developer's machine to have to rebuild for an hour to get them. This is part of the motivation for proposal package imports versioning in #46

@turadg before I merge, I realize I didn't address your feedback about the refactor this PR does. Are there still concerns with causing the images to rebuild on developers machines?

**Comment by turadg** [comment]
> Are there still concerns with causing the images to rebuild on developers machines?

Thanks for asking. It has the same impact but at this point I don't expect anyone to update to master on the local branches until refactoring to use the `@agoric/synthetic-chain` package (https://github.com/Agoric/agoric-3-proposals/pull/46/) so I don't think it's worth your time to back out the changes. They're also an improvement so I'll incorporate into the refactoring.

**Pull request merged**
